core/peerset/src/lib.rs

```diff
@@ -236,7 +236,9 @@ impl Peerset {
 		}
 
 		// Assign a slot for this reserved peer.
-		if let Some(pos) = self.data.out_slots.iter().position(|s| s.as_ref().map(|n| !self.data.reserved.contains(n)).unwrap_or(true)) {
+		// TODO: override slots that are occupied by not reserved peers
+		// send Message::Drop in those cases
+		if let Some(pos) = self.data.out_slots.iter().position(Option::is_none) {
```

Comment:
this is a bugfix. bootnodes were overwritten by reserved peers, but no message was created